### DISCUSSION TITLE
Prod permissions

### DISCUSSION THREAD

**Discussion by sebmr2** [mia-platform/community#583] [General]
### Description

Trying to prevent the CRUD service from having the dropCollection permission. When using the CRUD service to create views, it tries a drop and then creates the view as if it was always altering an existing view, My initial thought is I could elevate the accounts access and then remove post deploy. Will anything break if the service account has dropCollection removed? Is there a way to skip certain objects and generate a script for them?

**Comment by danibix95** [comment]
Hi @sebmr2, I created a MongoDB _custom role_ with limited privileges, so that I could test CRUD Service behavior. I observed that `dropCollection` privilege action is not needed by CRUD Service to create MongoDB views. In fact, all the functionalities seemed to continue working as intended. Consequently, you should _not_ need to add and then remove such permission from your Mongo user in order to operate CRUD Service.

Nonetheless, CRUD Service prints out a log every time an action it wants to perform cannot be performed due to insufficient privileges. These log state which command cannot be executed, easing the selection of which privilege actions should be applied. Therefore, you may further try experimenting with user privileges and observe which ones allows the service to work properly.

In the image below it can be seen which _privilege actions_ where assigned to the _custom role_ assigned to the Mongo user employed for the test I carried out.

![Screenshot from 2025-01-09 10-41-23](https://github.com/user-attachments/assets/65b84e6b-2b76-456c-a3e7-feb84156bcca)

**Reply by sebmr2** [reply]
readWrite would include dropCollection

**Reply by sebmr2** [reply]
My default user would normally have read, insert, update, and remove

**Reply by danibix95** [reply]
You're right. I updated the permissions list as follows, removing built-in roles:
```
find                        (all collections)
insert                      (all collections)
remove                      (all collections)
update                      (all collections)
createCollection            (all collections)
createIndex                 (all collections)
collMod                     (all collections)
dropIndex                   (all collections)
reIndex                     (all collections)
listIndexes                 (all collections)
bypassDocumentValidation    (all collections)
changeStream                (all collections)
validate                    (all collections)
listCollections             (all collections)
```
and experienced the mentioned startup problem when removing the `dropCollection` privilege.

In order to avoid requesting such permission, I updated the code to rather modify a view instead of dropping it, as it is shown [here](https://www.mongodb.com/docs/v7.0/core/views/update-view/). The change has been introduced with this [PR](https://github.com/mia-platform/crud-service/pull/439) and it can be tried out with latest tag `main` of CRUD Service.

**Reply by dallas-fletchall** [reply]
Thanks for all the hard work! 

I hate to ask for more, but is it possible to get a release candidate that includes this PR? 

![image](https://github.com/user-attachments/assets/2200d14b-2320-4160-95fa-da394f1b1f56).

We can build locally and perform some testing, but would feel more comfortable knowing that this will work in our environment if we could pull from a registry already setup in the cluster.

**Reply by danibix95** [reply]
Hi, you can now try out the RC version [`7.2.3-rc.0`](https://github.com/mia-platform/crud-service/pkgs/container/crud-service/336563521?tag=7.2.3-rc.0) of CRUD Service.

Let us know whether it works as expected.

**Reply by galatifederico** [reply]
Hi @dallas-fletchall, can you confirm that the version `7.2.3-rc.0` works as expected?
In that case we can proceed to create a tag with the new feature.

Thanks

**Comment by sebmr2** [comment]
Looks like everything is working. Much happier granting collMod vs dropCollection 😄

**Reply by dallas-fletchall** [reply]
Agree, tested with `7.2.3-rc.1`, with success.

**Reply by danibix95** [reply]
Hi! Thank you for your feedback. These changes will land soon in a new stable release

**Reply by danibix95** [reply]
Hi! New stable version [`v7.2.3`](https://github.com/mia-platform/crud-service/releases/tag/v7.2.3) of CRUD Service is now available.
